ui/src/shared/actions/annotations.ts

```diff
@@ -250,14 +254,20 @@ export const addAnnotationAsync = (
   annotation: Annotation
 ) => async dispatch => {
   dispatch(addAnnotation(annotation))
-  const savedAnnotation = await api.createAnnotation(createUrl, annotation)
-  dispatch(deleteAnnotation(annotation))
-  dispatch(addAnnotation(savedAnnotation))
+
+  try {
+    const savedAnnotation = await api.createAnnotation(createUrl, annotation)
+
+    dispatch(deleteAnnotation(annotation))
```

Comment:
are these two dispatches batched such that the user wouldn't see it disappear momentarily? i don't think they are but not 100% positive. what do you think about a single dispatch of `updateAnnotation` that replaces it with a single action? that's the pattern we've used previously as well.